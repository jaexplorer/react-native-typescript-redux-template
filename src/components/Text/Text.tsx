import React, { FC, ReactNode } from 'react';
import { Text as RNText, TextInputProps } from 'react-native';
import { Style } from '../../lib/Types';
import { useTheme } from '../../lib/Theme';
import stylesCreator from './TextStyles';

const Text: FC<TextProps> = ({ bold = false, children, style, ...props }) => {
  const [styles] = useTheme(stylesCreator, { bold });
  return (
    <RNText style={[styles.text, style]} maxFontSizeMultiplier={1} {...props}>
      {children}
    </RNText>
  );
};

export interface TextProps extends TextInputProps {
  bold?: boolean;
  children?: string | number | ReactNode[];
  style?: Style;
}

export default Text;
