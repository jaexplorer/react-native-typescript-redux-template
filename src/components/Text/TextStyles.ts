import { Platform, StyleSheet } from 'react-native';
import { StylesCreator } from '../../lib/Theme';

const stylesCreator: StylesCreator = (theme, scale, moderateScale, props) =>
  StyleSheet.create({
    text: {
      fontFamily: props.bold ? theme.font.bold : theme.font.regular,
      fontSize: props.bold ? moderateScale(14) : moderateScale(10),
      textAlignVertical: 'center',
      textAlign: 'center',
      bottom: Platform.select({ ios: -5, default: 0 }),
      color: props.bold ? theme.colours.textColours.secondary : theme.colours.textColours.primary,
      marginBottom: props.bold ? scale(2) : scale(0),
    },
  });

export default stylesCreator;
