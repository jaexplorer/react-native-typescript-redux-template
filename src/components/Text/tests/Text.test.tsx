import React from 'react';
import { render } from '@testing-library/react-native';
import Text from '../Text';

test('Does component render test', () => {
  const { baseElement } = render(<Text>Test</Text>);
  expect(baseElement).toMatchSnapshot();
});
