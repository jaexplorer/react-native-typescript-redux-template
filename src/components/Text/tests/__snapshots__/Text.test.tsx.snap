// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Does component render test 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      maxFontSizeMultiplier={1}
      style={
        Array [
          Object {
            "bottom": -5,
            "color": "#333333",
            "fontFamily": "",
            "fontSize": 15.714285714285714,
            "marginBottom": 0,
            "textAlign": "center",
            "textAlignVertical": "center",
          },
          undefined,
        ]
      }
    >
      Test
    </Text>
  </View>
</View>
`;
